bin/ch-convert

```diff
@@ -139,10 +140,48 @@ cv_docker_tar () {
     docker_ inspect "$1" \
             --format='{{range .Config.Env}}{{println .}}{{end}}' > "$tmpenv"
     tar rf "$tmptar" -b1 -P --xform="s|${tmpenv}|ch/environment|" "$tmpenv"
+
+    echo "adding metadata"
+    # shellcheck disable=SC2016
+    docker_ inspect "$1" \
+            --format='{{print "{\n"}}'\
+'{{printf "  \"arch\": %q,\n" .Architecture}}'\
+'{{print "  \"cwd\": "}}{{if .Config.WorkingDir}}'\
+'{{.Config.WorkingDir}}{{else}}{{print "\"\""}}{{end}}{{print ",\n"}}'\
+'{{print "  \"env\": {\n    "}}{{range $i, $v := .Config.Env}}'\
+'{{if $i}}{{print ",\n    "}}{{end}}{{printf "%q" $v}}{{end}}'\
+'{{print "\n  },\n"}}{{print "  \"labels\": {"}}'\
+'{{if .Config.Labels}}{{print "\n"}}'\
+'{{range $key, $v := .Config.Labels}}'\
+'{{printf "    \"%s\": \"%s\",\n" $key $v}}{{end}}{{print "  }"}}{{else}}'\
+'{{print "}"}}{{end}}{{print ","}}' > "$tmpmeta"
+
+# ContainerConfig.Cmd has the last command that was run in the image
+# We can use that to extract the shell that was used to build it.
+# As far as I can tell, that is always /bin/sh as with ch-image
+# Feels kludgy. But it gets the right result.
+last_cmd=$(docker_ inspect "$1" --format '{{json .ContainerConfig.Cmd}}')
+if echo "$last_cmd" | grep -q "\"/bin/sh\"\,\"\-c"; then
+    printf "  \"shell\": [\n    \"/bin/sh\",\n    \"-c\"\n  ],\n" >> "$tmpmeta"
+else
+    printf "  \"shell\": [\n    \"unknown\"\n  ],\n" >> "$tmpmeta"
+fi
```

Comment:
I think there's a non-kludgy way to get this, i.e., a direct key that gives the shell if it's not the default.

Reply:
That'd be great. I spent a bunch of time looking through the json files of various images, and this was the only common piece I could find.